presto-main/src/main/java/com/facebook/presto/operator/LookupSourceProvider.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import java.util.function.Function;
+
+public interface LookupSourceProvider
+        extends AutoCloseable
+{
+    <R> R withLease(Function<LookupSourceLease, R> action);
```

Comment:
Is there any reason to make this so abstract?  Why not just make this Page since that is the only user?

Reply:
You mean `R`? This won't be always a `Page`.